src/emdrv/gpioint.rs

```diff
@@ -1,19 +1,111 @@
-pub type IrqCallback = extern fn (pin: u8);
+use cmsis::nvic;
+use {gpio, int};
```

Comment:
I feel like the `int` name is kind of unfortunate and can lead to misunderstandings of what we're working with. What do you think?